src/save_options/registration.py

```diff
@@ -0,0 +1,111 @@
+import contextlib
+import inspect
+from collections.abc import Callable, Sequence
+from typing import Any
+
+from unrealsdk import logging
+
+from mods_base import BaseOption, GroupedOption, Mod, NestedOption
+from save_options.options import SaveOption
+
+# Allowing GroupedOption and NestedOption if all their children are SaveOption
+ModSaveOptions = dict[str, BaseOption]
+registered_save_options: dict[str, ModSaveOptions] = {}
+
+SaveCallback = Callable[[], None]
+save_callbacks: dict[str, SaveCallback] = {}
+
+LoadCallback = Callable[[], None]
+load_callbacks: dict[str, LoadCallback] = {}
+
+registered_mods: dict[str, Mod] = {}
+
+
+def _treat_as_save_option(obj: Any) -> bool:
+    if isinstance(obj, SaveOption):
+        return True
+    if isinstance(obj, GroupedOption | NestedOption):
+        if all(_treat_as_save_option(child) for child in obj.children):
+            return True
+        logging.error(
+            f"Option {obj.identifier} has both regular BaseOption and SaveOption"
+            f"defined as children. SaveOption instances will be ignored"  # noqa: COM812
+        )
+    return False
+
+
+def register_save_options(  # noqa: C901, D417
+    mod: Mod,
+    *,
+    save_options: Sequence[BaseOption] | None = None,
+    on_save: Callable[[], None] | None = None,
+    on_load: Callable[[], None] | None = None,
+    mod_identifier: str | None = None,
+) -> None:
```

Comment:
Should return the same mod instance which was passed, so you can chain it, same as `add_network_functions`.